src/PeriodDuration.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spatie\Period;
+
+use DateTimeImmutable;
+
+final class PeriodDuration
+{
+    /** @var Period */
+    private $period;
+
+    public function __construct(Period $period)
+    {
+        $this->period = $period;
+    }
+
+    public function equals(PeriodDuration $other): bool
+    {
+        return $this->startAndEndDatesAreTheSameAs($other)
+            || $this->includedStartAndEndDatesAreTheSameAs($other)
+            || $this->numberOfDaysIsTheSameAs($other)
+            || $this->compareTo($other) === 0;
+    }
+
+    public function isLargerThan(PeriodDuration $other): bool
+    {
+        return $this->compareTo($other) === 1;
+    }
+
+    public function isSmallerThan(PeriodDuration $other): bool
+    {
+        return $this->compareTo($other) === -1;
+    }
+
+    public function compareTo(PeriodDuration $other): int
+    {
+        $now = new DateTimeImmutable('@'.time()); // Ensure a TimeZone independent instance
+
+        $here = $this->period->getIncludedEnd()->diff($this->period->getIncludedStart(), true);
+        $there = $other->period->getIncludedEnd()->diff($other->period->getIncludedStart(), true);
+
+        return $now->add($here)->getTimestamp() <=> $now->add($there)->getTimestamp();
```

Comment:
Since we have access to `$here->period` and `$there->period`, what would you think about using the start dates of them instead of the absolute `$now` (which can be outside of the inspected periods and lead to edge cases that I can't see at the moment but am sure exist #leapyears 😅), for example something like:

```suggestion
        $here = $this->period->getIncludedEnd()->diff($this->period->getIncludedStart(), true);
        $there = $other->period->getIncludedEnd()->diff($other->period->getIncludedStart(), true);

        return $here->period->getIncludedStart()->add($here)->getTimestamp() <=> $there->period->getIncludedStart()->add($there)->getTimestamp();
```